bolt-design-system/bolt
# Fix: flag object's icon figure not rendering

## Jira

http://vjira2:8080/browse/BDS-931

## Summary

Fixed an issue with the figure component in a flag component not rendering icons.

## Details

The data structure with the refactored figure component does not play well with the how the flag object uses the figure component. So I made the following changes.

1. Updated the flag object to work with the refactored figure component.
2. Fixed all demo pages using the flag object with icon as the figure.
3. Created a condition to make the figure caption to render nothing if no data is passed.

## How to test

1. Run the branch locally.
2. Check the flags in these 2 pages, they are the icon + text layout in the 2nd band below the hero.
    * http://localhost:3000/pattern-lab/?p=pages-product-t2
    * http://localhost:3000/pattern-lab/?p=pages-product-landing
